### PR TITLE
[BEAM-13885] Add unit tests to window package

### PR DESCRIPTION
**Pull request by jrmccluskey**
Adds unit tests to window package, largely testing equality of windows and windowFns. Also adds function comments to the window Equals() functions that did not have them.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by jrmccluskey** [comment]
R: @damccorm @riteshghorse 

Somewhat boring test suite, just rolling through some equality test cases.

File: sdks/go/pkg/beam/core/graph/window/windows_test.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+			"global window == interval window[0,10]",
+			GlobalWindow{},
+			IntervalWindow{Start: 0, End: 10},
+			false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got, want := test.windowOne.Equals(test.windowTwo), test.expEquality; got != want {
+				t.Errorf("got (%v).Equals(%v) == %v, want %v", test.windowOne, test.windowTwo, got, want)
```
Comment (damccorm) [inline sdks/go/pkg/beam/core/graph/window/windows_test.go:72]: Small nit (here and elsewhere): the canonical/recommended pattern is function 1st, then got, then want - https://github.com/golang/go/wiki/TestComments#got-before-want - that's also what we use elsewhere AFIAK

File: sdks/go/pkg/beam/core/graph/window/fn_test.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			NewSessions(10 * time.Minute),
+			NewSessions(10 * time.Minute),
+			true,
+		},
+		{
+			"session inequal gap",
+			NewSessions(5 * time.Minute),
+			NewSessions(10 * time.Minute),
+			false,
+		},
```
Comment (damccorm) [inline sdks/go/pkg/beam/core/graph/window/fn_test.go:77]: Optional: Consider adding a mismatched type case (e.g. Global window vs Session).

**Review by damccorm — APPROVED** [review]
LGTM - left 2 tiny comments

File: sdks/go/pkg/beam/core/graph/window/fn_test.go (hunk 2 above)
Comment (jrmccluskey) [reply sdks/go/pkg/beam/core/graph/window/fn_test.go:77]: Good catch, added

File: sdks/go/pkg/beam/core/graph/window/windows_test.go (hunk 1 above)
Comment (jrmccluskey) [reply sdks/go/pkg/beam/core/graph/window/windows_test.go]: Good catch.

**Comment by riteshghorse** [comment]
add a blank line after PR description, otherwise it just bolds everything in description

**Comment by jrmccluskey** [comment]
I usually remember to, fixed

**Comment by riteshghorse** [comment]
LGTM

**Comment by jrmccluskey** [comment]
R: @lostluck  @youngoli

**Comment by jrmccluskey** [comment]
Run GoPortable PreCommit

**Review by lostluck — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**